vm/src/stdlib/itertools.rs

```diff
@@ -220,9 +220,13 @@ mod decl {
         }
 
         #[pymethod(magic)]
-        fn repr(&self) -> PyResult<String> {
-            let cur = self.cur.read();
-
+        fn repr(&self, vm: &VirtualMachine) -> PyResult<String> {
+            let mut cur = format!("{}", self.cur.read().clone().repr(vm)?);
+            let step = format!("{}", self.step.clone());
+            if step != "1" {
```

Comment:
I expected comparing the value instead of comparing the result.
Do you know what's the original way?

Reply:
cpython is also comparing the value (https://github.com/python/cpython/blob/3.10/Modules/itertoolsmodule.c#L4305-L4310).